### ISSUE TITLE
Upload is slower than bandwidth available

### ISSUE THREAD

**Issue by killforby** [Forceu/Gokapi#162]
Hello, having installed gokapi I find that when I upload the speed is not great between 10Mb and 12MB/s. Having done speed tests on my vps the speed is around 1175MB/s or 10 times more than what I can send with gokapi. The configuration is as follows: gokapi is installed on a vps and I access via a reverse proxy (NPM). Do you have any advice to recommend to me?

**Comment by killforby** [comment]
Hello, no we do not use encryption because I had read that it would slow down the speed. Would you have any advice to recommend to me?

**Comment by Forceu** [comment]
According to the speedtest, your connection would be a 10GBit connection, both at the receiving end, as well as at the sending end. Is that correct? Most providers only offer 100MBit or 1GBit for VPS instances. 100MBit would equal around 12MB/s, please check if that might be the case. Otherwise please upload a file with a different method (e.g. SSH) and compare the upload speed. Also please do a benchmark on how fast your VPS does SHA1 hashes. You can do this with the command `openssl speed sha1`, the last line will be your speed in kb/s.

**Comment by Forceu** [comment]
I will close this issue for now, as it seems to be a problem with your hardware instead of Gokapi. If you still experience the problem and it can be certain that it is Gokapi, please comment here so I can investigate further.

**Comment by killforby** [comment]
![image](https://github.com/Forceu/Gokapi/assets/94224508/f5200ae1-eb90-43e4-924e-088cc40ec0ad)
Good evening, I did the sha1 command as you asked me.

**Comment by killforby** [comment]
I performed a speedtest in cli
![image](https://github.com/Forceu/Gokapi/assets/94224508/f6db67a2-9722-4960-9145-2418201e7631)

**Comment by Forceu** [comment]
Hashing and bandwidth looks good. How long does it take to upload a file through SSH or FTP to the server?

**Comment by killforby** [comment]
hi with command dd =/if 
![image](https://github.com/Forceu/Gokapi/assets/94224508/bd010009-8eb5-4005-9819-860ca733f912)

With a 2g file on gokapi ( docker ) 6 minutes and 22sec

![image](https://github.com/Forceu/Gokapi/assets/94224508/d804a617-73f9-4f3d-99e5-5922a5357f5c)

**Comment by Forceu** [comment]
You are running all the tests on the server. Can you confirm that you are actually uploading the file to Gokapi from the same machine that runs Gokapi without any other device?

**Comment by killforby** [comment]
with fresh install 
![image](https://github.com/Forceu/Gokapi/assets/94224508/9e14d244-adda-496c-8e50-93221da65b9f)

**Comment by Forceu** [comment]
Yes, however you have to test the upload connection from your computer to the server. Try uploading a file to it from your computer to the server by other means, that will give you the speed that is possible.

**Comment by killforby** [comment]
![image](https://github.com/Forceu/Gokapi/assets/94224508/bf6d47a4-eeaa-4ac1-8ff3-4a99b6019cbd)
this is the download speed of my computer

**Comment by killforby** [comment]
On Filezilla 
![image](https://github.com/Forceu/Gokapi/assets/94224508/fc18e1a9-3bcd-4e6c-9e09-747eb2ca5ef9)

**Comment by Forceu** [comment]
Okay so if I read that right, the transfer speed from your computer to the server is 28MB/s?

**Comment by killforby** [comment]
Yes that's right, but for gokapi I don't exceed 12mb

**Comment by Forceu** [comment]
Does your connection have high latency (e.g. satellite connection)? In that case it might help if you increase the chunksize... 

In the main menu please open your browser console (Chrome: Shift + Ctrl + J) and enter the following:

`Dropzone.options.uploaddropzone["chunkSize"] = 50000000;`

Then - without reloading the page - try to upload something. Let me know if that increases the speed

**Comment by killforby** [comment]
Hello, no, I haven't had any noticeable changes. Wouldn't there be a variable to change in the .en for example?

**Comment by Forceu** [comment]
Okay, so I tested it with a 100MBit uplink to two different servers and could reproduce the problem. It might be related to chunking, which is done by Javascript. Unfortunately I am not sure yet what exactly is the cause and how to fix it. 

Does your speed increase if you enter these two commands in the console and then upload a file without refreshing the window?

```
dropzoneObject.options.chunkSize = 50000000;
dropzoneObject.options.parallelChunkUploads = true;
```

**Comment by killforby** [comment]
Do I enter the commands on Chrome or do I write them in the .env?

**Comment by killforby** [comment]
With the two commands entered on chrome I had a peak at 20mb/s

**Comment by Forceu** [comment]
Okay, so that's a good start. Do you get the same speeds if you only enter `dropzoneObject.options.chunkSize = 50000000;`?

**Comment by killforby** [comment]
When I run the command you gave me, I get 20mb/s but after 10 seconds the speed drops to 12mb/s

**Comment by killforby** [comment]
Thank you for being responsive and taking the time to address my concerns. There will be a new update

**Comment by Forceu** [comment]
Fixed in 3b7cdae. With that version it is not possible to set the chunksize and parallel upload limit in the settings. Default ist now 45MB chunk size and 4 parallel connections instead of 10MB serial.

Thank you very much for your feedback!
